src/qt/glcanvas.cpp

```diff
@@ -29,6 +24,9 @@ class wxQtGLWidget : public wxQtEventSignalHandler< QGLWidget, wxGLCanvas >
     wxQtGLWidget(wxWindow *parent, wxGLCanvas *handler, QGLFormat format)
         : wxQtEventSignalHandler<QGLWidget,wxGLCanvas>(parent, handler)
         {
+            setAttribute(Qt::WA_OpaquePaintEvent, true);
+            setAttribute(Qt::WA_NoSystemBackground, true);
+
```

Comment:
It would be more clear if you do this instead:
- Remove these lines
```suggestion
```
- and in `wxGLCanvas::Create()`, line 495, put this
```
    if ( !wxWindow::Create( parent, id, pos, size, style, name ) )
        return false;

    SetBackgroundStyle(wxBG_STYLE_PAINT);

    return true;

```

Reply:
`wxBG_STYLE_PAINT` doesn't seem like it enables `WA_NoSystemBackground`

Reply:
> `wxBG_STYLE_PAINT` doesn't seem like it enables `WA_NoSystemBackground`

Sorry, but why do we need this attribute here? what kind of problems does this solve?

Reply:
Qt changes the clear color and clears the canvas, also I think it disables depth testing.

This causes wrong background color and disabled depth testing in opengl samples, because they only set this state in InitGL.

I tried removing one of the flag setters, but it didn't work correctly.

Reply:
I couldn't see this happening in any of the samples!

Anyway, `WA_OpaquePaintEvent` according to Qt docs:

> Indicates that the widget paints all its pixels when it receives a paint event. Thus, it is not required for operations like updating, resizing, scrolling and focus changes to erase the widget before generating paint events.

So I don't see the benefit that `WA_NoSystemBackground` adds with `WA_OpaquePaintEvent` attribute ?

You may find [this](https://lists.qt-project.org/pipermail/interest/2015-January/014903.html) useful!

But if you still see that we need both for everything to work properly, then let's do it like this.

Reply:
Will have to double-check tomorrow.

Reply:
So turns out WA_NoSystemBackground wasn't needed.